client/asset/zec/zec.go

```diff
@@ -285,14 +294,207 @@ func connect(c rpcCaller) error {
 	if err := createAccount(0); err != nil {
 		return err
 	}
-	// When shielded pools are implemented, we'll use a separate, dedicated
-	// account that uses unified addresses with only and Orchard receiver type.
-	// if err := createAccount(1); err != nil {
-	// 	return err
-	// }
+	if err := createAccount(1); err != nil {
+		return err
+	}
 	return nil
 }
 
+func (w *zecWallet) lastShieldedAddress() (addr string, err error) {
+	if addrPtr := w.lastAddress.Load(); addrPtr != nil {
+		return addrPtr.(string), nil
+	}
+	accts, err := zListAccounts(w)
+	if err != nil {
+		return "", err
+	}
+	for _, acct := range accts {
+		if acct.Number != shieldedAcctNumber {
+			continue
+		}
+		if len(acct.Addresses) == 0 {
+			break // generate first address
+		}
+		lastAddr := acct.Addresses[len(acct.Addresses)-1].UnifiedAddr
+		w.lastAddress.Store(lastAddr)
+		return lastAddr, nil // Orchard = Unified for account 1
+	}
+	return w.NewShieldedAddress()
+}
+
+// ShieldedBalance list the last address and the balance in the shielded
+// account.
+func (w *zecWallet) ShieldedStatus() (status *asset.ShieldedStatus, err error) {
+	// z_listaccounts to get account 1 addresses
+	// DRAFT NOTE: It sucks that we need to list all accounts here. The zeroth
+	// account is our transparent addresses, and they'll all be included in the
+	// result. Should probably open a PR at zcash/zcash to add the ability to
+	// list a single account.
+	status = new(asset.ShieldedStatus)
+	status.LastAddress, err = w.lastShieldedAddress()
+	if err != nil {
+		return nil, err
+	}
+
+	status.Balance, err = zGetBalanceForAccount(w, shieldedAcctNumber)
+	if err != nil {
+		return nil, err
+	}
+
+	return status, nil
+}
+
+// Balance adds a sum of shielded pool balances to the transparent balance info.
+func (w *zecWallet) Balance() (*asset.Balance, error) {
+	bal, err := w.ExchangeWalletFullNode.Balance()
```

Comment:
Wallets that were used before `getnewaddress` was deprecated will have addresses in a "legacy" account. `getbalance` returns balance of **all** transparent outputs, regardless of account, and including the "legacy" account. I wouldn't even know how to get just the legacy account balance to add to the `z_getbalanceforaccount` result for transparent account 0. Similarly, `listunspent` returns all transparent outputs, regardless of account. The only reason we need an account 0 is because `getnewaddress` is deprecated. I've taken care that addresses produced from the shielded account 1 don't include a transparent receiver, so barring external RPC sabotage, all t addrs will be from legacy or 0. 

Side note: It is unfortunate that you even need to include a shielded address in a unified address, but zcashd requires it, so our account 0 addresses **do** have a shielded receiver, though neither the unified address or the Orchard receiver are ever exposed to the caller.

> EDIT: Oh, I see your impl of zGetBalanceForAccount only gets Orchard balance, ignoring transparent value.

Generalizing that return output would be no problem. I just didn't have a reason yet.